src/enc.cpp

```diff
@@ -87,12 +90,22 @@ int emain()
     }
 
     const Memfs memfs(_memfs_name);
+
+    _log_verbose("invoking premain");
+
+    // Copy potentially existing payload data into string (for null-termination)
+    // and pass it to ego's premain
+    const auto payload_data_pair = ert::payload::get_data();
+    std::string payload_data(
+        static_cast<char const*>(payload_data_pair.first),
+        payload_data_pair.second);
+
+    ert_ego_premain(&_argc, &_argv, payload_data.c_str());
```

Comment:
```suggestion
    // Copy potentially existing payload data into string (for null-termination)
    // and pass it to ego's premain
    const auto payload_data_pair = payload::get_data();
    const string payload_data(
        static_cast<const char*>(payload_data_pair.first),
        payload_data_pair.second);

    _log_verbose("invoking premain");
    ert_ego_premain(&_argc, &_argv, payload_data.c_str());
```
find all differences :D